tutorials/time-freq/plot_ssvep.py

```diff
@@ -0,0 +1,698 @@
+mne.viz.plot_topomap(snrs_12hz_chaverage, topo_pos_grave, vmin=1., axes=ax)
+
+print("sub 2, 12hz trials, SNR at 12hz")
+print("average SNR (all channels): %f" % snrs_12hz_chaverage.mean())
+print("average SNR (occipital ROI): %f" % snrs_target.mean())
+
+tstat_roi_vs_scalp = \
+    ttest_rel(snrs_target.mean(axis=1), snrs_12hz.mean(axis=1))
+print("12 hz SNR in occipital ROI is significantly larger than 12 hz SNR over "
+      "all channels: t = %.3f, p = %f" % tstat_roi_vs_scalp)
+
+##############################################################################
+# We can see, that 1) this participant indeed exhibits a cluster of channels
+# with high SNR in the occipital region and 2) that the average SNR over all
+# channels is smaller than the average of the visual ROI computed above.
+# The difference is statistically significant. Great!
+#
+# Such a topography plot can be a nice tool to explore and play with your data
+# - e.g. you could try how changing the reference will affect the spatial
+# distribution of SNR values.
+#
+# However, we also wanted to show this plot to point at a potential
+# problem with frequency-tagged (or any other brain imaging) data:
+# there are many channels and somewhere you will likely find some
+# statistically significant effect.
+# It is very easy - even unintended - to end up double-dipping or p-hacking.
+# So if you want to work with an ROI or individual channels, ideally select
+# them *a priori* - before collecting or looking at the data - and preregister
+# this decision so people will believe you.
+# If you end up selecting an ROI or individual channel for reporting *because
+# this channel or ROI shows an effect*, e.g. in an explorative analysis, this
+# is also fine but make it transparently and correct for multiple comparison.
+#
+# Statistical analysis
+# --------------------
+# Back from this little detour into open science, let's move on and
+# do the analyses we actually wanted to do:
+#
+# We will show that we can easily discriminate the brains responses in the
+# trials with different stimulation frequencies.
+#
+# In the frequency and SNR spectrum plot above, we had all trials mixed up.
+# Now we will extract 12 and 15 hz SNR as well as SNR of the 1st and 2nd
+# harmonic in both types of trials individually, and compare the values with
+# a simple t-test.
+#
+
+snrs_roi = snrs[:, picks_roi_vis, :].mean(axis=1)
+
+freq_plot = [12, 15, 24, 30, 36, 45]
+color_plot = [
+    'darkblue', 'darkgreen', 'mediumblue', 'green', 'blue', 'seagreen'
+]
+xpos_plot = [-5. / 12, -3. / 12, -1. / 12, 1. / 12, 3. / 12, 5. / 12]
+fig, ax = plt.subplots()
+labels = ['12hz trials', '15hz trials']
+x = np.arange(len(labels))  # the label locations
+width = 0.6  # the width of the bars
+res = dict()
+
+# loop to plot SNRs at stimulation frequencies and harmonics
+for i, f in enumerate(freq_plot):
+    # extract snrs
+    stim_12hz_tmp = \
+        snrs_roi[i_trial_12hz, np.argmin(abs(np.subtract(freqs, f)))][0]
```

Comment:
you don't need to use np.subtract everywhere. Just use `-` you work with array here

Reply:
done